Source/Core/VideoCommon/AbstractTexture.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2009 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+#include "VideoCommon/VideoCommon.h"
+
+class AbstractTexture
+{
+public:
+  explicit AbstractTexture(const TextureConfig& c)
+    : config(c)
+  {}
+
+  virtual ~AbstractTexture()
+  {}
+
+  virtual void Bind(unsigned int stage) = 0;
+  virtual bool Save(const std::string& filename, unsigned int level) = 0;
+  
+  virtual void CopyRectangleFromTexture(const AbstractTexture* source,
+    const MathUtil::Rectangle<int>& srcrect,
+    const MathUtil::Rectangle<int>& dstrect) = 0;
+  virtual void Load(const u8* buffer, u32 width, u32 height, u32 expanded_width,
+    u32 level) = 0;
+  virtual void FromRenderTarget(bool is_depth_copy, const EFBRectangle& srcRect, bool scaleByHalf,
```

Comment:
Not sure what you mean?  I pulled this API directly from the class that already exists, this just moves it into a new location (and tidies up a few things)